### PR TITLE
Hotfix/filter subscription videos

### PR DESCRIPTION
**Pull request by wickloww**
## Description

Fixes filter on subscription videos and adds tests

## Has this been tested?

- [x] 👍 yes, I added tests to the test suite
- [ ] 💭 no, because this PR is a draft and still needs work
- [ ] 🙅 no, because this PR does not update server code
- [ ] 🙋 no, because I need help

### CODE REVIEW

File: server/tests/api/videos/videos-common-filters.ts [hunk 1]
```diff
@@ -162,17 +176,21 @@ describe('Test videos filter', function () {
...
+    async function getVideosNames (
+      options: {
+        server: PeerTubeServer
+        isLocal?: boolean
+        include?: VideoInclude
+        privacyOneOf?: VideoPrivacy[]
+        token?: string
+        expectedStatus?: HttpStatusCode
+      },
+      skipSubscription = false
```
Comment (Chocobozzz) [inline server/tests/api/videos/videos-common-filters.ts:188]: Why not put `skipSubscription` in `options`?

File: server/tests/api/videos/videos-common-filters.ts [hunk 2]
```diff
@@ -162,17 +176,21 @@ describe('Test videos filter', function () {
...
+    ) {
       const videosResults: string[][] = []
 
       for (const path of paths) {
+        if (skipSubscription && path === subscriptionVideosPath) continue
```
Comment (Chocobozzz) [inline server/tests/api/videos/videos-common-filters.ts:193]: Please add a `\n` at the end of the line, to separate cases where we skip the loop with the loop content

File: server/tests/api/videos/videos-common-filters.ts [hunk 3]
```diff
@@ -70,10 +76,18 @@ describe('Test videos filter', function () {
 
   describe('Check deprecated videos filter', function () {
 
-    async function getVideosNames (server: PeerTubeServer, token: string, filter: string, expectedStatus = HttpStatusCode.OK_200) {
+    async function getVideosNames (
```
Comment (Chocobozzz) [inline server/tests/api/videos/videos-common-filters.ts:79]: Please convert all args as an object

When you reach 3 or 4 arguments it means it's time to switch to the object arg style

File: server/tests/api/videos/videos-common-filters.ts (hunk 1 above)
Comment (wickloww) [reply server/tests/api/videos/videos-common-filters.ts]: Correct in https://github.com/Chocobozzz/PeerTube/pull/5665/commits/cee2a3fc25413173a465eb2392681605261cff78.

**Pull request merged**